crates/s3s/src/response.rs

```diff
@@ -17,13 +17,16 @@ pub struct S3Response<T> {
 }
 
 impl<T> S3Response<T> {
-    pub fn new(output: T) -> Self {
+    pub fn new_with_headers(output: T, headers: HeaderMap<HeaderValue>) -> Self {
         Self {
             output,
-            headers: HeaderMap::new(),
+            headers,
             extensions: Extensions::new(),
         }
     }
+    pub fn new(output: T) -> Self {
+        Self::new_with_headers(output, HeaderMap::new())
+    }
```

Comment:
I prefer the following order and style
1. the major constructor `new`, and then additional constructors `with_*`
2. a blank line between each method
3. trivial constructors do not depend on each other

```rust
    pub fn new(output: T) -> Self {
        Self {
            output,
            headers: HeaderMap::new(),
            extensions: Extensions::new(),
        }
    }

    pub fn with_headers(output: T, headers: HeaderMap<HeaderValue>) -> Self {
        Self {
            output,
            headers,
            extensions: Extensions::new(),
        }
    }
```